### PR TITLE
Avoid formatting disk multiple times in fio

### PR DESCRIPTION
**Pull request by ehankland**
@noahl @yuyantingzero  Could you please take a look?

### CODE REVIEW

File: perfkitbenchmarker/benchmark_spec.py [hunk 1]
```diff
@@ -157,7 +157,7 @@ def UnPickleLock(locked, *args):
     'os_type is "rhel". In general if two OS\'s use the same package manager, '
     'and are otherwise very similar, the same os_type should work on both of '
     'them.')
-flags.DEFINE_string('scratch_dir', '/scratch',
+flags.DEFINE_string('scratch_dir', None,
```
Comment (noahl) [inline perfkitbenchmarker/benchmark_spec.py]: Does this mean that *every* benchmark now gets its disk unmounted by default?

File: perfkitbenchmarker/benchmarks/fio_benchmark.py [hunk 2]
```diff
@@ -352,20 +358,15 @@ def Prepare(benchmark_spec):
...
-
-  logging.info('Umount scratch disk on %s at %s', vm, mount_point)
-  vm.RemoteCommand('sudo umount %s' % mount_point)
 
   if FLAGS.against_device or FLAGS.prefill_device:
     logging.info('Fill device %s on %s', disk.GetDevicePath(), vm)
     FillDevice(vm, disk, FLAGS.device_fill_size)
 
-  if not FLAGS.against_device:
+  if not FLAGS.against_device and FLAGS.prefill_device:
```
Comment (noahl) [inline perfkitbenchmarker/benchmarks/fio_benchmark.py]: We need to mount the disk if we're not running against the device, regardless of whether we're prefilling it.

File: perfkitbenchmarker/benchmark_spec.py (hunk 1 above)
Comment (ehankland) [inline perfkitbenchmarker/benchmark_spec.py]: No - the default is to use the value in default_config_constants.yaml, which is still /scratch.

File: perfkitbenchmarker/benchmarks/fio_benchmark.py (hunk 2 above)
Comment (ehankland) [inline perfkitbenchmarker/benchmarks/fio_benchmark.py]: If we are not prefilling it, then it is already mounted (it is only not mounted if the against_device or prefill_device flags are specified).

Comment (noahl) [inline perfkitbenchmarker/benchmarks/fio_benchmark.py]: Got it.

**Comment by noahl** [comment]
LGTM.

**Pull request merged**